src/assets/TokensController.ts

```diff
@@ -547,16 +547,12 @@ export class TokensController extends BaseController<
     }
   }
 
-  async _createEthersContract(
+  _createEthersContract(
```

Comment:
I just want to say, I don't agree with what we are doing here. This method should not exist, and I don't think we should be stubbing `ethers.Contract`. I think it would be better to use dependency injection and have TokensController take a Contract constructor which can be swapped out at runtime for something else. That would obviate the need for this method. I realize we can't really change this right now given that there is a test in the extension which relies on stubbing this method, but I think that would be the next logical move here.

Reply:
agreed, the whole thing is anti-patterns on anti-patterns. But its okay. At this point, anything that helps towards resolving test non-determinism is good. The next refactor will be a little bit smaller and easier. Many more small improvements will make the big improvement PRs much less busy.

Reply:
Awesome, look forward to it!